File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03147/p03147_num0_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> heights(N);
    for (int i = 0; i < N; i++) {
        cin >> heights[i];
    }

    int operations = 0;
    for (int i = 1; i < N; i++) {
        if (heights[i] > heights[i - 1]) {
            operations += heights[i] - heights[i - 1];
        }
    }

    cout << operations << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03147/p03147_num0_parsed.py
'''
# Helper function to find the minimum number of watering operations
def minimum_watering_operations(N, heights):
    operations = 0
    
    # Iterate through each flower
    for i in range(N):
        if i == 0:
            operations += heights[i]
        else:
            if heights[i] > heights[i-1]:
                operations += heights[i] - heights[i-1]
    
    return operations

# Read input from standard input
N = int(input())
heights = list(map(int, input().split()))

# Calculate the minimum number of watering operations
min_operations = minimum_watering_operations(N, heights)

# Print the result
print(min_operations)
'''